docs/examples/data_transfer_objects/defining_dtos_on_layers.py

```diff
@@ -1,31 +1,53 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
 from typing import List
 from uuid import UUID, uuid4
 
 from litestar import Controller, delete, get, post, put
+from litestar.app import Litestar
+from litestar.dto import DataclassDTO
+from litestar.dto.config import DTOConfig
+
+
+@dataclass
+class User:
+    name: str
+    email: str
+    age: int
+    id: UUID = field(default_factory=uuid4)
+
 
-from .models import User, UserDTO, UserReturnDTO
+class UserWriteDTO(DataclassDTO[User]):
+    config = DTOConfig(exclude={"id"})
+
+
+class UserReadDTO(DataclassDTO[User]): ...
 
 
 class UserController(Controller):
-    dto = UserDTO
-    return_dto = UserReturnDTO
+    dto = UserWriteDTO
+    return_dto = UserReadDTO
 
-    @post()
+    @post("/", sync_to_thread=False)
     def create_user(self, data: User) -> User:
         return data
 
-    @get()
+    @get("/", sync_to_thread=False)
     def get_users(self) -> List[User]:
-        return [User(id=uuid4(), name="Litestar User")]
+        return [User(name="Mr Sunglass", email="mr.sunglass@example.com", age=30)]
 
-    @get("/{user_id:uuid}")
+    @get("/{user_id:uuid}", sync_to_thread=False)
     def get_user(self, user_id: UUID) -> User:
-        return User(id=user_id, name="Litestar User")
+        return User(id=user_id, name="Mr Sunglass", email="mr.sunglass@example.com", age=30)
 
-    @put("/{user_id:uuid}")
+    @put("/{user_id:uuid}", sync_to_thread=False)
     def update_user(self, data: User) -> User:
         return data
 
-    @delete("/{user_id:uuid}", return_dto=None)
+    @delete("/{user_id:uuid}", return_dto=None, sync_to_thread=False)
     def delete_user(self, user_id: UUID) -> None:
         return None
+
+
+app = Litestar(UserController)
```

Comment:
shouldn't this be?
```suggestion
app = Litestar([UserController])
```

Reply:
seems its related to concurrency issues, https://github.com/litestar-org/litestar/pull/3382 disabled concurrency and the tests ran fine